demo/src/main/java/eu/kevin/demo/countryselection/CountrySelectionViewModel.kt

```diff
@@ -0,0 +1,99 @@
+package eu.kevin.demo.countryselection
+
+import androidx.lifecycle.AbstractSavedStateViewModelFactory
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import androidx.savedstate.SavedStateRegistryOwner
+import eu.kevin.common.architecture.BaseViewModel
+import eu.kevin.common.entities.LoadingState
+import eu.kevin.demo.ClientProvider
+import eu.kevin.demo.countryselection.entities.Country
+import eu.kevin.demo.countryselection.usecases.SupportedCountryUseCase
+import eu.kevin.demo.routing.DemoRouter
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+internal class CountrySelectionViewModel constructor(
+    private val countryUseCase: SupportedCountryUseCase,
```

Comment:
dependency naming should be full - `supportedCountryUseCase` (maybe even rename this to `GetSupportedCountriesUseCase`